source/common/stats/stats_impl.h

```diff
@@ -333,27 +340,50 @@ class GaugeImpl : public Gauge, public MetricImpl {
   // Stats::Gauge
   virtual void add(uint64_t amount) override {
     data_.value_ += amount;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void dec() override { sub(1); }
   virtual void inc() override { add(1); }
   virtual void set(uint64_t value) override {
     data_.value_ = value;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void sub(uint64_t amount) override {
     ASSERT(data_.value_ >= amount);
     ASSERT(used());
     data_.value_ -= amount;
   }
   virtual uint64_t value() const override { return data_.value_; }
-  bool used() const override { return data_.flags_ & RawStatData::Flags::Used; }
+  bool used() const override { return data_.flags_ & Flags::Used; }
 
 private:
   RawStatData& data_;
   RawStatDataAllocator& alloc_;
 };
 
+/**
+ * Structure to hold statistical data of histogram.
+ */
+struct HistogramStatistics {
+public:
+  HistogramStatistics() {}
+
+  HistogramStatistics(histogram_t* histogram_ptr) {
+    hist_approx_quantile(histogram_ptr, const_cast<double*>(quantiles_in_),
+                         ARRAY_SIZE(quantiles_in_), quantiles_out_);
+  }
+
+  std::string summary() const {
+    return fmt::format("P0: {} , P25: {}, P50: {}, P75: {}, P90: {}, P95: {}, P99: {}, P100: {}",
+                       quantiles_out_[0], quantiles_out_[1], quantiles_out_[2], quantiles_out_[3],
+                       quantiles_out_[4], quantiles_out_[5], quantiles_out_[6], quantiles_out_[7],
+                       quantiles_out_[8]);
+  }
+
+  double quantiles_in_[9] = {0, 0.25, 0.5, 0.75, 0.90, 0.95, 0.99, 0.999, 1};
```

Comment:
These should all be static constants. If you can't figure out how to do the C++ I think either @mrice32 or @jmarantz can hopefully help.